x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/frozen_indices_handling/trial_license_complete_tier/frozen_indices_handling.ts

```diff
@@ -0,0 +1,204 @@
+  const createTestRule = async (indices: string[]) => {
+    const createRuleProps: CreateRuleProps = {
+      body: {
+        name: `spammy_rule`,
+        description: 'Spammy query rule',
+        enabled: true,
+        risk_score: 1,
+        rule_id: 'rule-1',
+        severity: 'low',
+        type: 'query',
+        query: 'foo: *',
+        index: indices,
+        from: 'now-3600s',
+      },
+    };
+
+    const { body: createdRuleResponse } = await securitySolutionApi
+      .createRule(createRuleProps)
+      .expect(200);
+
+    return createdRuleResponse.id;
+  };
+
+  const getRuleExecutionResult = async (ruleId: string) => {
+    const startDate = new Date(Date.now() - 60 * 60 * 1000).toISOString();
+    const ruleExecutionsResultUrl = `/internal/detection_engine/rules/${ruleId}/execution/results`;
+    return await supertest
+      .get(`${ruleExecutionsResultUrl}?start=${startDate}&end=9999-12-31T23:59:59Z`)
+      .set('elastic-api-version', '1')
+      .send()
+      .expect(200);
+  };
+
+  describe('@ess frozen indices queried', () => {
+    let cleanupStack: CleanupStack;
+    let ruleId: string;
+    beforeEach(async () => {
+      cleanupStack = getCleanupStack();
+    });
+
+    afterEach(async () => {
+      await cleanupStack.runCleanup();
+      await deleteAllAlerts(supertest, log, es);
+      await deleteAllRules(supertest, log);
+    });
+
+    describe('when an index that is not in the frozen data tier is queried', () => {
+      const index = `test_idx-${uuidv4()}`;
+
+      it('should not report that frozen indices were queried', async () => {
+        await indexSampleData(index);
+
+        ruleId = await createTestRule(['test_idx*']);
+
+        // Wait for a successful execution. The counter should be zero at this point
+        await retry.try(async () => {
+          const response = await getRuleExecutionResult(ruleId);
+          expect(response.statusCode).to.be.equal(200);
+          const { total, events } = response.body;
+          expect(total).to.be.equal(1);
+          expect(events[0].status).to.be.equal('success');
+          expect(events[0].frozen_indices_queried_count).to.be.equal(0);
+        });
+
+        cleanupStack.push(async () => {
+          await es.indices.delete({
+            index,
+          });
+        });
+      });
+    });
+
+    describe('when an index that is in the frozen data tier is queried', () => {
+      const index = `test_idx-${uuidv4()}`;
+
+      it('should report that frozen indices were queried', async () => {
+        await indexSampleData(index);
+
+        ruleId = await createTestRule(['test_idx*']);
+
+        const { newIndexName, snapshotRepositoryName, ilmPolicyName } =
+          await moveIndexToFrozenDataTier({ es, index, retry });
+
+        // schedule a rule run after moving the index to frozen
+        await supertest
+          .post(`/internal/alerting/rule/${ruleId}/_run_soon`)
+          .set('kbn-xsrf', 'foo')
+          .expect(204);
```

Comment:
We can create the rule at this point instead of creating it earlier and calling run_soon

Reply:
Done